### PR TITLE
Add admin parent name to analysis table

### LINKED ISSUE #356: 'Run analysis' results should show each admin level's name
When we run an analysis on an admin 2, or admin 3 (etc.) level dataset, we should display the parent admin level's names in the table as well.

### PR DESCRIPTION
**Pull request by tdlinh2712**
This closes #356 
deploy: http://prism-issue-356.surge.sh/
modified the Name column in analysis table to include parent admin level's names. This also makes the name matches location name on tooltip
image

### CODE REVIEW

File: src/context/analysisResultStateSlice.ts [hunk 1]
```diff
@@ -126,7 +126,9 @@ function generateTableFromApiData(
     );
 
     const name: string =
-      featureBoundary?.properties?.[adminLevelName] || 'No Name';
+      adminLayer.adminLevelNames
```
Comment (ericboucher) [inline src/context/analysisResultStateSlice.ts:129]: Let's create a util function so that the tooltip and table can reuse the same and stay in sync

**Comment by wadhwamatic** [comment]
This looks good, but two questions come up:
1. How will this work in combination with PR 391 https://github.com/WFP-VAM/prism-frontend/pull/391? I assume we're sorting on the the lowest admin level (admin 3), but this will look rather confusing for a user. It may make more sense to include each name level as a separate column instead to make sorting more clear
2. It seems inconsistent to have local name only display the lowest level while English shows multiple levels. However, since we're working on localization separate (PR 382 https://github.com/WFP-VAM/prism-frontend/pull/382) it seems we can just drop local names in this view if the localized version will include this information anyway. @ericboucher - thoughts?

**Comment by tdlinh2712** [comment]
@wadhwamatic 
to answer q1: with the current solution, the name will be sorted alphabetically based on the highest admin level name. If two locations share the same level 1 name, then it would be sorted based on level 2 name, and so on. 
for example:

Kayah, Loikaw, Hpruso
Shan (North), Lashio, Hseni
Shan (North), Lashio, Kunlong
Shan (South), Loilen, Laihka
Shan (South), Loilen, Mongkaing

I don't think this is bad since we can see all locations of the same parent admin-level together

**Comment by tdlinh2712** [comment]
@wadhwamatic @ericboucher deploy link updated

**Comment by wadhwamatic** [comment]
> @wadhwamatic to answer q1: with the current solution, the name will be sorted alphabetically based on the highest admin level name. If two locations share the same level 1 name, then it would be sorted based on level 2 name, and so on. for example:
> 
> I don't think this is bad since we can see all locations of the same parent admin-level together

@tdlinh2712 - agreed, this works fine for us. This is good from my side, but pending final code review from @ericboucher or @Ry-DS 

@ericboucher - note 'local name' in boundary files needs to be thought through as part of the localization work

File: src/utils/name-utils.ts [hunk 2]
```diff
@@ -0,0 +1,13 @@
+import { get } from 'lodash';
+import { BoundaryLayerProps } from '../config/types';
+
+export function getLocationName(
+  layer: BoundaryLayerProps,
+  featureBoundary: any,
```
Comment (Ry-DS) [inline src/utils/name-utils.ts:6]: Should avoid `any` use at all costs. Can't you use `Feature<Geometry, GeoJsonProperties>`? If there's a type we can't match it should be defined manually or with `unknown`

File: src/utils/name-utils.ts [hunk 3]
```diff
@@ -0,0 +1,13 @@
+import { get } from 'lodash';
+import { BoundaryLayerProps } from '../config/types';
+
+export function getLocationName(
```
Comment (Ry-DS) [inline src/utils/name-utils.ts:4]: This function should have a docstring
`/**
...
**/`

File: src/context/analysisResultStateSlice.ts [hunk 4]
```diff
@@ -125,8 +126,7 @@ function generateTableFromApiData(
       feature => feature.properties?.[adminLevelName] === row[adminLevelName],
     );
 
-    const name: string =
-      featureBoundary?.properties?.[adminLevelName] || 'No Name';
+    const name: string = getLocationName(adminLayer, featureBoundary);
```
Comment (Ry-DS) [inline src/context/analysisResultStateSlice.ts:129]: specifying the type here seems uneccessary (from the previous code but we could remove it now)
```suggestion
    const name = getLocationName(adminLayer, featureBoundary);
```

File: src/utils/name-utils.ts (hunk 3 above)
Comment (Ry-DS) [inline src/utils/name-utils.ts:4]: I like `getFullName` better than `getLocationName` thoughts?

File: src/utils/name-utils.ts (hunk 2 above)
Comment (tdlinh2712) [reply src/utils/name-utils.ts]: thanks! done

File: src/utils/name-utils.ts (hunk 3 above)
Comment (tdlinh2712) [reply src/utils/name-utils.ts]: done. Can you check the docstring to make sure what i wrote makes sense?

Comment (tdlinh2712) [reply src/utils/name-utils.ts]: changed to `getFullLocationName`

File: src/utils/name-utils.ts [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+/**
+ * Format full location name in ascending admin levels
+ * @param layer admin layer, both props and data
+ * @param featureBoundary geospatial features of the location, contains names of different admin layers
+ * @returns a string containing location name in ascending admin levels i.e level 1 name, level 2 name, etc
+ */
+
+export function getFullLocationName(
+  layer: BoundaryLayerProps,
+  featureBoundary:
```
Comment (Ry-DS) [inline src/utils/name-utils.ts:14]: Use optional `featureBoundary?: <type>` instead of an explicit `undefined`

Comment (tdlinh2712) [reply src/utils/name-utils.ts]: done

**Review by Ry-DS — APPROVED** [review]
Looks good!

File: src/components/MapView/Layers/BoundaryLayer/index.tsx [hunk 6]
```diff
@@ -30,9 +30,7 @@ function BoundaryLayer({ layer }: { layer: BoundaryLayerProps }) {
 
   const onClickFunc = (evt: any) => {
     const coordinates = evt.lngLat;
-    const locationName = layer.adminLevelNames
-      .map(level => get(evt.features[0], ['properties', level], '') as string)
-      .join(', ');
+    const locationName = getFullLocationName(layer, evt.features[0]);
```
Comment (ericboucher) [inline src/components/MapView/Layers/BoundaryLayer/index.tsx:33]: I think we only need to pass `layer.adminLevel`

**Pull request merged**